src/connectors/mssql/connector.ts

```diff
@@ -0,0 +1,205 @@
+          reject(error);
+        } else {
+          success = true;
+        }
+      });
+      
+      const parameterKeys = Object.keys(_parameters);
+      for (const key of parameterKeys) {
+        const parameter = _parameters[key];
+        
+        request.addParameter(
+          parameter.name,
+          parameter.type,
+          parameter.value
+        );
+      }
+      
+      const rows: unknown[] = [];
+      request.on('row', (columns = []) => {
+        const currentRow = {};
+        for (const column of columns) {
+          const { value, metadata } = column;
+          const { colName } = metadata;
+          
+          currentRow[colName] = value;
+        }
+        
+        rows.push(currentRow);
+      });
+      
+      request.on('requestCompleted', () => {
+        connection.close();
+        resolve({ rows, success });
+      });
+      
+      request.on('error', (error) => {
+        connection.close();
+        reject(error);
+      });
+      
+      connection.execSql(request);
+    });
+    
+    try {
+      const {
+        rows,
+        success,
+      } = await query;
+      
+      return {
+        rows,
+        success,
+      };
+    } catch (error) {
+      error.sql = _sql;
+      error.parameters = parameters;
+      console.error(error);
+    }
+  }
+  
+  return <Connector>{
+    name: "mssql",
+    dialect: "mssql",
+    exec(sql: string) {
+      return _run(sql, []);
+    },
+    prepare(sql: string) {
+      const statement = <Statement>{
+        _sql: sql,
+        _params: [],
+        bind(...params) {
+          if (params.length > 0) {
+            this._params = params;
+          }
+          return statement;
+        },
+        async all(...params) {
+          const { rows } = await _run(this._sql, params || this._params);
+          return rows;
+        },
+        async run(...params) {
+          const { success = false } = await _run(this._sql, params || this._params) || {};
+          return {
+            success,
+          };
+        },
+        async get(...params) {
+          const { rows: [ row ] } = await _run(this._sql, params || this._params);
+          return row;
+        },
+      };
+      
+      return statement;
+    },
+  };
+};
+
+// taken from the `kysely` library: https://github.com/kysely-org/kysely/blob/413a88516c20be42dc8cbebade68c27669a3ac1a/src/dialect/mssql/mssql-driver.ts#L440
+export function getTediousDataType(value: unknown): DataType {
+  if (value === null || value === undefined || typeof value === 'string') {
+    return TYPES.NVarChar;
+  }
+  
+  if (typeof value === 'bigint' || (typeof value === 'number' && value % 1 === 0)) {
+    return value < -2_147_483_648 || value > 2_147_483_647 ? TYPES.BigInt : TYPES.Int;
+  }
+  
+  if (typeof value === 'number') {
+    return TYPES.Float;
+  }
+  
+  if (typeof value === 'boolean') {
+    return TYPES.Bit;
+  }
+  
+  if (value instanceof Date) {
+    return TYPES.DateTime;
```

Comment:
There is also a DateTime2 type in tedious and in SQL

Reply:
@peterbud This code was taken directly from the Kysely library, but upon research I do see that `DateTime2` is a more suitable suggestion for anything new going forward. The [MSSQL documentation](https://learn.microsoft.com/en-us/sql/t-sql/data-types/datetime2-transact-sql?view=sql-server-ver16) says that it aligns with the existing SQL standard and should be able to be used as a drop-in replacement.

Would you prefer if we use DateTime2 as the default here instead?

Reply:
I think that would be better. datetime has issues with fractional milliseconds.